api/src/email/route.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  Type,
+  type FastifyPluginCallbackTypebox
+} from '@fastify/type-provider-typebox';
+import { isPast } from 'date-fns';
+
+let fiveMinuteLimit: Date;
+
+export const settingRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.addHook('onRequest', fastify.authenticateSession);
+  fastify.addHook('onResponse', async (_request, _reply) => {
+    if (fiveMinuteLimit !== undefined || isPast(fiveMinuteLimit))
+      return (fiveMinuteLimit = new Date(new Date().getTime() + 5 * 60 * 1000));
+  });
+
+  fastify.put(
+    '/update-my-email',
+    {
+      schema: {
+        body: Type.Object({
+          email: Type.String({ format: 'email' })
+        }),
+        response: {
+          200: Type.Object({
+            message: Type.Literal('flash.email-valid'),
+            type: Type.Literal('success')
+          }),
+          204: Type.Object({
+            message: Type.String(),
+            type: Type.Literal('info')
+          }),
```

Comment:
I don't see where we're actually sending a 204 status, but my understanding is that HTTP responses with a 204 status should not have a body.

Reply:
Oliver is taking care of the PR, because of how complex rx is, and I can't avoid it